src-java/kilda-pce/src/test/java/org/openkilda/pce/finder/DiversityPathFinderTest.java

```diff
@@ -393,6 +393,91 @@ public void shouldChooseLongerPath() throws RecoverableException, UnroutableFlow
         );
     }
 
+    @Test
+    public void choosePathWithoutSingleSwitchFlowTest() throws RecoverableException, UnroutableFlowException {
+        // Topology:
+        // -----E----D
+        // |         |
+        // A----B----C     Already created single switch flow: E
+        //                 Requested flow: A-B-C-D
+
+        List<IslImmutableView> isls = new ArrayList<>();
+        isls.addAll(getBidirectionalIsls(switchA, 1, switchB, 2));
+        isls.addAll(getBidirectionalIsls(switchB, 3, switchC, 4));
+        isls.addAll(getBidirectionalIsls(switchC, 5, switchD, 6));
+        isls.addAll(getBidirectionalIsls(switchD, 7, switchE, 8));
+        isls.addAll(getBidirectionalIsls(switchE, 9, switchA, 10));
+
+        FlowPath forwardPath = FlowPath.builder()
+                .srcSwitch(switchE)
+                .destSwitch(switchE)
+                .pathId(FORWARD_PATH_ID)
+                .segments(Collections.emptyList())
+                .build();
+        when(flowPathRepository.findById(FORWARD_PATH_ID)).thenReturn(java.util.Optional.of(forwardPath));
```

Comment:
just curious why the fully qualified name is used in this test? Any conflict with some other Optional?

Reply:
Good catch, thanks! Will fix it.